### PR TITLE
[MAIN-2105] switch related_pods from hikaru to k8s api

### PR DESCRIPTION
**Pull request by aantn**
Needs more testing prior to merging

### CODE REVIEW

**Comment by aantn** [comment]
@Avi-Robusta please also note that our test framework is failing on this, which means I might have broken something for the crashpod playbook. Make sure you get the tests working before merging.

**Comment by Avi-Robusta** [comment]
tested discovery nodes isn't changed
- [ ] compared in stg my the same cluster with previous runner version robusta `upgrade cluster `to `upgrade cluster-performance`

File: src/robusta/patch/patch.py [hunk 1]
```diff
@@ -59,6 +63,46 @@ def patched_setter(self, on_pod_conditions):
...
+    Fix based on files at end of https://github.com/tomplus/kubernetes_asyncio/pull/300/files
+    """
+    try:
+        setattr(Configuration, 'get_default', classmethod(get_default))
+    except Exception as e:
+        logging.error(f"Failed to add 'get_default' method to Configuration: {e}")
+        return
+
+    # Loop over all model classes in the kubernetes.client.models package
+    for name, obj in inspect.getmembers(kubernetes.client.models):
```
Comment (aantn) [inline src/robusta/patch/patch.py:78]: How long does it take to run this on startup?

File: playbooks/robusta_playbooks/k8s_resource_enrichments.py [hunk 2]
```diff
@@ -71,82 +90,122 @@ class RelatedPod(BaseModel):
...
-        pods = PodList.listNamespacedPod(namespace=resource.metadata.namespace, label_selector=selector).obj.items
-
-    return pods
-
-
-def to_pod_obj(pod: Pod, cluster: str) -> RelatedPod:
-    resource_requests = pod_requests(pod)
-    resource_limits = pod_limits(pod)
-    addresses = ",".join([address.ip for address in pod.status.podIPs])
+        logging.error(f"start listing pods for selector: {selector};")
```
Comment (aantn) [inline playbooks/robusta_playbooks/k8s_resource_enrichments.py:155]: Can you remove this log please? (Sorry, I know I added it!)

File: playbooks/robusta_playbooks/k8s_resource_enrichments.py [hunk 3]
```diff
@@ -188,12 +247,20 @@ def related_pods(event: KubernetesResourceEvent, params: RelatedPodParams):
...
     Supports Deployments, ReplicaSets, DaemonSets, StatefulSets and Pods
     """
-    pods = get_related_pods(event.get_resource())
     cluster = event.get_context().cluster_name
 
     if params.output_format == "json":
-        event.add_enrichment([JsonBlock(json.dumps([to_pod_obj(pod, cluster).dict() for pod in pods]))])
+        pods = get_related_pods(event.get_resource(), limit=params.limit)
+        event.add_enrichment([JsonBlock(json.dumps([to_pod_obj(pod, cluster, include_raw_data=params.include_raw_data).dict() for pod in pods], default=str))])
+    elif params.output_format == "json2":
```
Comment (aantn) [inline playbooks/robusta_playbooks/k8s_resource_enrichments.py:255]: I know I added it, but you can remove this json2 code! It wasn't used in the end.

File: src/robusta/core/discovery/discovery.py [hunk 4]
```diff
@@ -523,9 +525,11 @@ def discovery_process() -> DiscoveryResults:
             )
             raise e
         Discovery.stacktrace_thread_active = False
+
+        # very likely it is nodes that are the problem and we can fix by converting here away from k8s format
```
Comment (aantn) [inline src/robusta/core/discovery/discovery.py:529]: Please remove the comment (sorry, I am responsible for this one too!)

File: src/robusta/core/sinks/robusta/robusta_sink.py [hunk 5]
```diff
@@ -418,8 +354,9 @@ def __publish_new_nodes(self, current_nodes: V1NodeList, node_requests: Dict[str
 
         # new or changed nodes
         for node_name in curr_nodes.keys():
-            pod_requests = node_requests.get(node_name, [])  # if all the pods on the node have no requests
-            updated_node = self.__from_api_server_node(curr_nodes.get(node_name), pod_requests)
+            # if all the pods on the node have no requests
+            updated_node = curr_nodes.get(node_name)
+            # TODO: uodateRequests
```
Comment (aantn) [inline src/robusta/core/sinks/robusta/robusta_sink.py:359]: yeah, do this :)

**Review by aantn — COMMENTED** [review]
Left some comments!

**Review by Avi-Robusta — APPROVED** [review]

**Pull request merged**